optaplanner-core/src/main/java/org/optaplanner/core/config/AbstractConfig.java

```diff
@@ -51,6 +54,15 @@
      */
     public abstract Config_ copyConfig();
 
+    /**
+     * Call the class visitor on each (possibly null) referenced class of this config (including those referenced by children
+     * config).
+     * Required to create the bean factory in Quarkus.
+     *
+     * @return Set of classes referenced by this config
+     */
+    public abstract void visitReferencedClasses(Consumer<Class<?>> classVisitor);
```

Comment:
Nice solution to the problem!
I'd be a bit more descriptive in the Javadoc, though - what does it mean for a class to be "referenced"? Say something to the tune of "each Class instance provided to this config by the user".